### PR TITLE
Janitors cleaning the floor now generates 50 techweb points per tile cleaned that has anything to clean

### PR DESCRIPTION
**Pull request by silicons**
Now you have a better reason to be making yourself a horrible, slippery nuisance! 
:cl:
rscadd: Mops now have adaptive sensors on them that can generate research points when cleaning up messes! Unfortunately, this only works if the user is a trained and skilled janitor, and not some random with a mop...
experimental: Mops store the points generated. They can be uploaded to the station's research databanks by using them on a R&D console.
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
I'm scared that this will become the only useful point generation method. Also does it count spray cleaner bottles? They are nerfed.

**Comment by Cyberboss** [comment]
Without feedback to the janitor, this isn't worth merging

**Comment by kevinz000** [comment]
@Shadowflame909 this only works for mops, I feel like allowing cleaner turf reactions to do that would be too much buffing

File: code/game/objects/items/mop.dm [hunk 1]
```diff
@@ -24,16 +26,20 @@
...
+	var/cleaned = FALSE
 	if(reagents.has_reagent("water", 1) || reagents.has_reagent("holywater", 1) || reagents.has_reagent("vodka", 1) || reagents.has_reagent("cleaner", 1))
 		SEND_SIGNAL(A, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_MEDIUM)
 		for(var/obj/effect/O in A)
 			if(is_cleanable(O))
+				cleaned = TRUE
 				qdel(O)
+	if(cleaned && user && user.mind && user.mind.assigned_role == "Janitor")
+		SSresearch.station_tech.add_points_all(CLEAN_TILE_REWARD)
+		to_chat(user, "<span class='notice'>Your [src] flashes a message that it has gained useful information from eradicating the mess on the floor. Good work.</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/mop.dm]: Too spammy, maybe just like a CoD style +10 popup only the janitor can see

Comment (XDTM) [reply code/game/objects/items/mop.dm]: I suddenly want a janitor point system

Comment (kevinz000) [reply code/game/objects/items/mop.dm]: @Cyberboss Hmmmm what if I incorporated it into the existing "you clean the floor" message ?

**Comment by Trilbyspaceclone** [comment]
TRILBY -Janitor sci catgirl, bomber, rnd, RD, robotics

**Comment by Pyko1** [comment]
The new meta is to lock up a puking clown and the janitor in 3x3 area?

**Comment by epochayur** [comment]
is this a joke PR?

**Comment by Mickyan** [comment]
I like the concept but I really don't like the idea of encouraging players to do their job in the worst, most disruptive way possible

At the very least the spray bottle and especially the floor buffer should still generate points even if at a much lower rate per tile compared to mopping. Give RnD incentive to research it as soon as possible.

**Comment by Kromgar** [comment]
Is this a meme pr?

**Comment by kevinz000** [comment]
Alright fine I'll are cleaner and buffer if possible when I'm home

**Comment by Wilchenx** [comment]
>kevinz runs out of ideas how to make techwebs more engaging

**Comment by epochayur** [comment]
oh no he's serious

**Comment by ExcessiveUseOfCobblestone** [comment]
Put it on the advanced mop to encourage people to rush janitor tech

**Comment by Wilchenx** [comment]
can't wait for all the wet tiles

**Comment by Iatots** [comment]
Now everyone will call me a powergamer just because I like the mop.
Or at least they would, if I played.

**Comment by Arianya** [comment]
Learn to W A L K
The janitor is meant to be somewhat inconvenient for the crew. Deal with it.
>b-but my floor buffer and space cleaner
I swear to god I will throw you out an airlock

I'm somewhat whatever on research point generation by itself, since janitor generally doesn't care about RP, but I could see some nice impact for it.

@kevinz000 if you wanted to create a "light" economy, maybe make the RP be stored in a datastick in the mop, which has to be handed over to R&D? Increases interactivity and gives R&D a reason to research service related techs in exchange for the datastick.

**Comment by kevinz000** [comment]
:+1:

**Comment by ExcessiveUseOfCobblestone** [comment]
It doesn't make sense at all for the normal mop though, this should be exclusively to the advanced one.

**Comment by Repukan** [comment]
Janitor req is a must

**Comment by pubby** [comment]
Points aren't rewards and don't make a job more rewarding. It's the things you can buy with the points that's rewarding.

Arianya was on the right track when they suggested a bartering system.

**Comment by Repukan** [comment]
Points are rewards and any retard would do any action if you gave them meaningless points each time they did it.

I.E. Sissyphus, Cookie Clicker

**Comment by Tlaltecuhtli** [comment]
is money  a reward or what you buy with money a reward? 🤔

**Comment by davethwave** [comment]
I was thinking that perhaps the scientists could do puzzles and stuff as "research" with each puzzle giving more tech points depending on difficulty and your completion time.

**Comment by Mickyan** [comment]
@Arianya considering how many tools the janitor gets that don't make people slip that might just be your own opinion

If you're doing the datastick thing then you can forget space cleaner but the floor buffer definitely should as it's an RnD item and it would be the best tool for janitors to both do their job properly and help the research department in return for it

**Comment by Arianya** [comment]
@Mickyan The fact that people have slipped in tools to mitigate the inconvenience doesn't change the original intent of the role. Were it within my power I would move space cleaner out of being there at roundstart and make the janitor order a pimpin' ride before he could put a floor buffer on it, but I'm not given sole creative vision on the game ;)

**Comment by Metroidlover** [comment]
Couldn't they just use space cleaner instead of water on the mop and not be a nuisance?

**Comment by Shadowflame909** [comment]
Slipping is super lame. As long as this doesn't end up replacing automatic point generation. I will always space the mop and glass table any janitor who likes to clean public hallways with water.

**Comment by kevinz000** [comment]
also no this is not my solution for techwebs this is something I threw together with 5 minutes of spare time to make a fun feature for janitors involving techwebs. Going to say this before a certain someone comes in screaming that I'm avoiding the issues by outsourcing research to people mopping the floors.

**Comment by Wilchenx** [comment]
>before a certain someone comes

what is this damage control

**Comment by CosmicScientist** [comment]
Archaic concern: role restriction, though minor and plenty of other roles have their own special interactions, this means the clown can no longer ~~slip and space nuke ops~~ slip the crew *whilst* earning points. I unno, is this still an unwritten rule?

I do want to see HoPs open fifteen janitor slots just to power through science!

File: code/game/objects/items/mop.dm [hunk 2]
```diff
@@ -24,15 +27,30 @@
...
+	var/cleaned = FALSE
 	if(reagents.has_reagent("water", 1) || reagents.has_reagent("holywater", 1) || reagents.has_reagent("vodka", 1) || reagents.has_reagent("cleaner", 1))
 		SEND_SIGNAL(A, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_MEDIUM)
 		for(var/obj/effect/O in A)
 			if(is_cleanable(O))
+				cleaned = TRUE
 				qdel(O)
+	if(cleaned && user && user.mind && user.mind.assigned_role == "Janitor")
+		stored_points += CLEAN_TILE_REWARD
+		to_chat(user, "<span class='notice'>Your [src] flashes a message that it has gained useful information from eradicating the mess on the floor. Good work.</span>")
```
Comment (CosmicScientist) [inline code/game/objects/items/mop.dm]: Could you only alert me once I reach a threshold? I clean five tiles at a time.

**Comment by lolman360** [comment]
when are we gonna do something like /vg/'s tetris RnD

**Comment by Shadowflame909** [comment]
when oranges likes the idea

**Comment by kevinz000** [comment]
message condensed

**Comment by CosmicScientist** [comment]
Huzzah!

**Comment by Bawhoppen** [comment]
Still waiting for tech points to be generated by having people cuffed in brig cells.

**Comment by fludd12** [comment]
Wasn't there that one idea where brig cells have treadmills that generate power or tech points or something?

**Comment by bgobandit** [comment]
in before people meta it by beating the shit out of and repeatedly healing a random lizard/assistant/clown, then cleaning their blood

which actually is kind of very ss13 if you think about it

**Comment by Fox-McCloud** [comment]
I can see science non-ironically asking janitor to be borrowed for a short period of time, only for the mto repeatedly spam a mess creating object on the floor/beating a monkey/making blood just to generate points.

This is an *interesting* idea, but a horrendously exploitable one.

**Comment by Iamgoofball** [comment]
@bgobandit people set up monk dojos in the chapel where assistants volunteered to get beaten to near death in exchange for monk powers when we had D&D monks

**Comment by PKPenguin321** [comment]
I won't lie, this is definitely me when I murder the janitor

**Comment by epochayur** [comment]
has anyone reiterated how dumb of an idea this is

**Comment by Militaires** [comment]
new meta: flyman janitor eating shit then cleaning up his puke 200 times until he gets a phazon

**Comment by Cyberboss** [comment]
As funny as this is, and as much as I want to reward the janitor for doing his job, I honestly don't think techweb points are the way to do it

**Comment by adamkad1** [comment]
but science loving janitor would be fun

**Pull request closed without merging**